cpp/src/Ice/Instance.cpp

```diff
@@ -873,11 +883,9 @@ namespace
 }
 
 InstancePtr
-IceInternal::Instance::create(const Ice::CommunicatorPtr& communicator, const Ice::InitializationData& initData)
+IceInternal::Instance::create(const Ice::InitializationData& initData)
 {
-    auto instance = shared_ptr<Instance>(new Instance(initData));
-    instance->initialize(communicator);
-    return instance;
+    return shared_ptr<Instance>(new Instance(initData));
```

Comment:
The previous code looks more elegant - why this update?

Also, since the only caller (Communicator) is a friend, it doesn't need this create function.

Reply:
We create the SSLEngine in Instance::initialize, the communicator passed to `Instance::initialize` doesn't have a _instance pointer set. We pass this communicator to SSLEngine and when the engine tries to use the communicator it crash because the instance is not set.

I thinking now that might be simpler to pass the Istance to the SSLEngine, or the properties and the logger which seems that it is all it needs.

Reply:
Reworked to keep the previous code.